languagetool-server/src/main/java/org/languagetool/server/HTTPServerConfig.java

```diff
@@ -237,6 +238,10 @@ private void parseConfigFile(File file, boolean loadLangModel, boolean loadWord2
         if (cacheSize < 0) {
           throw new IllegalArgumentException("Invalid value for cacheSize: " + cacheSize + ", use 0 to deactivate cache");
         }
+        if (props.containsKey("cacheTTLSeconds") && !props.containsKey("cacheSize")) {
+          System.err.println("Use of cacheTTLSeconds without also setting cacheSize has no effect.");
```

Comment:
As warnings tend to get  ignored, I'd be fine to just throw an exception here.

Reply:
done